evm/gas-service.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+const { ethers } = require('hardhat');
+const { getDefaultProvider, Contract } = ethers;
+const { Command, Option } = require('commander');
+const {
+    printInfo,
+    printWalletInfo,
+    printWarn,
+    mainProcessor,
+    prompt,
+    getContractJSON,
+    getGasOptions,
+    wasEventEmitted,
+    isValidAddress,
+    validateParameters,
+    httpPost,
+    loadConfig,
+} = require('./utils');
+const { addBaseOptions } = require('./cli-utils');
+const { getWallet } = require('./sign-utils');
+
+async function getGasUpdates(env, chain, destinationChains) {
```

Comment:
Should probably add this check here as well since you export this function to be used in other files:
```
validateParameters({
                isNonEmptyStringArray: { destinationChains },
});
```